rpc/metrics.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package rpc
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/metrics"
+)
+
+var (
+	jsonrpcRequestGauge    = metrics.NewRegisteredGauge("rpc/jsonrpc/count/in", nil)
+	successfulRequestGauge = metrics.NewRegisteredGauge("rpc/jsonrpc/count/success", nil)
+	failedReqeustGauge     = metrics.NewRegisteredGauge("rpc/jsonrpc/count/failure", nil)
+	jsonrpcServingTimer    = metrics.NewRegisteredTimer("rpc/jsonrpc/duration/all", nil)
+)
+
+func newJsonrpcServingTimer(version string, method string, valid bool) metrics.Timer {
+	flag := "success"
+	if !valid {
+		flag = "failure"
+	}
+	m := fmt.Sprintf("rpc/jsonrpc/duration/%s/%s/%s", version, method, flag)
```

Comment:
The `version` here -- isn't that just always `2.0`, the json rpc version?

Reply:
I may be paranoid here, but it seems like this could blow up our metrics, if we allow callers to spin up new meters arbitrarily. Could we either

- Limit the amount of meters to some max, or, 
- Whitelist the potential meters, so we don't spin up meters against methods that do not exist?

Reply:
I remove the `version`. It seems useless.

And regarding the limitation, if the method user specified is non-existent, then the request won't be served and a `methodNotFoundError` will be returned.

So what metrics we will spin up here are: `available_methods_success` or `available_methods_failure`. It should be limited